crypto/digestible/derive/src/lib.rs

```diff
@@ -82,6 +78,126 @@ fn try_digestible(input: TokenStream) -> Result<TokenStream, &'static str> {
     Ok(expanded.into())
 }
 
+fn try_digestible_enum(
+    ident: &Ident,
+    generics: &Generics,
+    variant_data: &DataEnum,
+) -> Result<TokenStream, &'static str> {
+    let call = variant_data
+        .variants
+        .iter()
+        .map(|variant| {
+            let variant_ident = &variant.ident;
+
+            // Our behavior differs based on whether the enum variant is a unit (has no data
+            // associated with it), or named (has sruct data associated with it), or unnamed (has
+            // tuple data assocated with it).
+            match &variant.fields {
+                // For an enum variant that doesn't have associated data (e.g. SomeEnum::MyVariant)
+                // we generate code that looks like this:
+                // Self::MyVariant => {
+                //   hasher.input("MyVariant").as_bytes());
+                // }
+                Fields::Unit => {
+                    quote! {
+                        Self::#variant_ident => {
+                            hasher.input(stringify!(#variant_ident).as_bytes());
+                        },
+                    }
+                }
+
+                // For an enum variant that has anonymous fields (e.g. SomeEnum::MyVariant(u32,
+                // u64)) we generate code that looks like this:
+                // Self::MyVariant(field_0, field_1) => {
+                //   hasher.input("MyVariant").as_bytes());
+                //   hasher.input("0").as_bytes());
+                //   field_0.digest(hasher);
+                //   hasher.input("1").as_bytes());
+                //   field_1.digest(hasher);
+                // }
+                Fields::Unnamed(FieldsUnnamed {
+                    unnamed: fields, ..
+                }) => {
+                    let field_idents = fields
+                        .iter()
+                        .enumerate()
+                        .map(|(idx, _field)| format_ident!("field_{}", idx))
+                        .collect::<Vec<_>>();
+
+                    let per_field_digest = fields
+                        .iter()
+                        .enumerate()
+                        .map(|(idx, _field)| {
+                            let index = syn::Index::from(idx);
+                            let field_ident = format_ident!("field_{}", idx);
+                            quote! {
+                                hasher.input(stringify!(#index).as_bytes());
```

Comment:
i think for the proof to apply, you want to turn `index` into a fixed-width representation like u32 or something.
in early reviews of the proof we proposed the size can be different per enum and argued about how that should work. but its really a very small detail. but I think it is important that there is some fixed size per enum, so that representation of 10'th member has the same length as the 2nd member

Reply:
never mind, I'm not looking at the right part here, sorry